client/src/components/Signup.js

```diff
@@ -0,0 +1,89 @@
+import React, { useState } from 'react';
+import '../loginAndSignup.css';
+import { Link } from 'react-router-dom';
+
+async function createUser(username, password) {
+  const response = await fetch('/api/users', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      username,
+      password,
+    }),
+  });
+  const data = await response.json();
```

Comment:
Make sure you check `response.ok` before getting the json. If you don't, you will miss most of the backend errors.